mcs/class/System.Core/System.Security.Cryptography.X509Certificates/DSACertificateExtensions.cs

```diff
@@ -0,0 +1,64 @@
+//
+// DSACertificateExtensions.cs
+//
+// Authors:
+//	Filip Navara <filip.navara@gmail.com>
+//
+// Copyright (C) 2018 .NET Foundation and Contributors
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+namespace System.Security.Cryptography.X509Certificates
+{
+	public static class DSACertificateExtensions
```

Comment:
Could you use the CoreFX version from `src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/DSACertificateExtensions.cs` and stub the PAL bits with NIE instead

Reply:
(Now I see the formatting came up messed up, will fix that.)

The reason I didn't use the CoreFX version was that there are two other similar classes (for ECDsa and RSA) that also had their implementations in Mono. It would make sense to replace them all at the same time, but that would require bigger refactoring.